internal/resources/metal/metal_ssh_key/resource.go

```diff
@@ -52,21 +52,21 @@ func metalSSHKeyCommonFields() map[string]*schema.Schema {
 	}
 }
 
-func resourceMetalSSHKey() *schema.Resource {
+func Resource() *schema.Resource {
 	return &schema.Resource{
-		Create: resourceMetalSSHKeyCreate,
-		Read:   resourceMetalSSHKeyRead,
-		Update: resourceMetalSSHKeyUpdate,
-		Delete: resourceMetalSSHKeyDelete,
+		Create: ResourceMetalSSHKeyCreate,
+		Read:   ResourceMetalSSHKeyRead,
+		Update: ResourceMetalSSHKeyUpdate,
+		Delete: ResourceMetalSSHKeyDelete,
 		Importer: &schema.ResourceImporter{
 			State: schema.ImportStatePassthrough,
 		},
 
-		Schema: metalSSHKeyCommonFields(),
+		Schema: MetalSSHKeyCommonFields(),
 	}
 }
 
-func resourceMetalSSHKeyCreate(d *schema.ResourceData, meta interface{}) error {
+func ResourceMetalSSHKeyCreate(d *schema.ResourceData, meta interface{}) error {
```

Comment:
nit: We shouldn't need to export the CRUD functions, just the Resource or DataSource itself.

Reply:
And since we are sharing CRUD and Schema between the project ssh key and user ssh keys, we could still use the exported Resource as the base and then give it a Project variation of Schema.  The shared schema (`MetalSSHKeyCommonFields`) could be exported.